src/filters/i18n-validation-exception.filter.ts

```diff
@@ -28,37 +28,53 @@ export class I18nValidationExceptionFilter implements ExceptionFilter {
       detailedErrors: true,
     },
   ) {}
+
   catch(exception: I18nValidationException, host: ArgumentsHost) {
     const i18n = I18nContext.current();
 
     const errors = formatI18nErrors(exception.errors ?? [], i18n.service, {
       lang: i18n.lang,
     });
 
+    const normalizedErrors = this.normalizeValidationErrors(errors);
+
     switch (host.getType() as string) {
       case 'http':
         const response = host.switchToHttp().getResponse();
+
+        const responseBody = this.buildResponseBody(
+          host,
+          exception,
+          normalizedErrors,
+        );
+
         response
           .status(this.options.errorHttpStatusCode || exception.getStatus())
-          .send({
-            statusCode:
-              this.options.errorHttpStatusCode || exception.getStatus(),
-            message: exception.getResponse(),
-            errors: this.normalizeValidationErrors(errors),
-          });
+          .send(responseBody);
         break;
       case 'graphql':
-        exception.errors = this.normalizeValidationErrors(
-          errors,
-        ) as I18nValidationError[];
+        exception.errors = normalizedErrors as I18nValidationError[];
         return exception;
     }
   }
 
-  private isWithErrorFormatter(
-    options: I18nValidationExceptionFilterOptions,
-  ): options is I18nValidationExceptionFilterErrorFormatterOption {
-    return 'errorFormatter' in options;
+  protected buildResponseBody(
+    host: ArgumentsHost,
+    exc: I18nValidationException,
+    errors: string[] | I18nValidationError[] | object,
+  ) {
+    if ('responseBodyFormatter' in this.options) {
+      return this.options.responseBodyFormatter(host, exc, errors);
+    } else {
```

Comment:
redundant else , can be ommited as the control flow would naturally return the block

Reply:
add eslint rule if you think that's good, I just prefer explicit if else to be honest